### PR TITLE
doc: Added User-defined Git Repo

### PR DESCRIPTION
**Pull request by ashokdevtron**
- 'GitOps' doc in Global Configurations has been revamped
- 'GitOps Configuration' doc has been added in App Configurations
- Included RBAC
- Replaced old snapshots
- Linked with 'Clone Application' doc and 'CD Workflow' doc
- Added Gitops for Helm Apps
- Added feature flag with steps

### CODE REVIEW

File: docs/user-guide/creating-application/gitops-config.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Users need to have [Admin permission](../../user-guide/global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to configure user-defined Git repo.
+{% endhint %}
+
+1. Go to **Applications** (choose your app) → **App Configuration** (tab) → **GitOps Configuration**.
+
+    ![Figure 1: App-level GitOps Config](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/gitops/app-config-gitops.jpg)
+
+2. Assuming a GitOps repo was not added to your application earlier, you get 2 options:
+
+    * **Auto-create repository** - Select this option if you wish to proceed with the default behavior. This will auto-create a repository whose name will be the same as the application name.
```
Comment (abhibhaw) [inline docs/user-guide/creating-application/gitops-config.md:26]: With some prefix...

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Missing GitOps configuration for helm apps

File: docs/user-guide/cloning-application.md [hunk 2]
```diff
@@ -21,3 +21,7 @@ Now click on `Clone App` to clone the selected application.
 
 New application with a duplicate template is created.
 
+{% hint style="warning" %}
+In case the application you are cloning had a [GitOps Configuration](./creating-application/gitops-config.md), it won't be copied to your cloned application.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/cloning-application.md]: Fixed

File: docs/user-guide/creating-application/gitops-config.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# GitOps Configuration
+
+{% hint style="warning" %}
+The 'GitOps Configuration' page appears only if the super-admin has enabled 'Allow changing git repository for application' in [Global Configurations → GitOps](../global-configurations/gitops.md).
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/creating-application/gitops-config.md:4]: Noted. Will add it once it is finalized.

[the remark this answers is not in the document]
Comment (abhibhaw) [reply docs/user-guide/creating-application/gitops-config.md:4]: `FEATURE_USER_DEFINED_GITOPS_REPO_ENABLE=false`
`FEATURE_USER_DEFINED_GITOPS_REPO_ENABLE=true`
@ashokdevtron

[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/creating-application/gitops-config.md:4]: Thanks, done 👍

File: docs/user-guide/creating-application/gitops-config.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+### Feature Flag
+
+* **For disabling**: `FEATURE_USER_DEFINED_GITOPS_REPO_ENABLE=false`
```
Comment (abhibhaw) [inline docs/user-guide/creating-application/gitops-config.md:8]: We also need to tell that this feature flag needs to go in ConfigMap of dashboard of the devtron instance. Maybe Attach an image of resource browser?

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: docs/user-guide/creating-application/gitops-config.md (hunk 4 above)
Comment (ashokdevtron) [reply docs/user-guide/creating-application/gitops-config.md]: Done. Added steps and a gif in GC's GitOps doc and removed from here.

**Review by abhibhaw — APPROVED** [review]
LGTM

**Pull request merged**